src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/DisasterRecoveryFileValidator.cs

```
@@ -0,0 +1,153 @@
+
+                foreach (string filePath in args.FilePaths)
+                {
+                    bool IsFolder = false;
+                    bool Existing = IsPathExisting(serverConnection, filePath, ref IsFolder);
+
+                    if (Existing)
+                    {
+                        if (IsFolder)
+                        {
+                            errorMessage = string.Format(SR.BackupPathIsFolderError, filePath);
+                            result = false;
+                            break;
+                        }
+                    }
+                    else
+                    {
+                        // If the file path doesn't exist, check if the folder exists
+                        string folderPath = PathWrapper.GetDirectoryName(filePath);
+                        if (isLocal)
+                        {
+                            if (!string.IsNullOrEmpty(folderPath) && !Directory.Exists(folderPath))
+                            {
+                                errorMessage = string.Format(SR.InvalidBackupPathError, folderPath);
+                                result = false;
+                                break;
+                            }
+                        }
+                        else
+                        {
+                            bool isFolderOnRemote = true;
+                            bool existsOnRemote = IsPathExisting(serverConnection, folderPath, ref isFolderOnRemote);
+                            if (!existsOnRemote)
+                            {
+                                errorMessage = string.Format(SR.InvalidBackupPathError, folderPath);
+                                result = false;
+                                break;
+                            }
+                        }
+                    }
+                }
+            }
+            else
+            {
+                result = false;
+            }
+
+            if (!string.IsNullOrEmpty(errorMessage))
```

>errorMessage [](start = 38, length = 12)

you don't need to return the result if you always throw exception for invalid paths

---

changed to send error message instead of throwing exception